packages/monitor-v2/src/monitor-polymarket/MonitorProposalsOrderBook.ts

```diff
@@ -1,116 +1,190 @@
-  const livePolymarketProposalRequests = [...proposedPriceRequestsOOv2, ...proposedPriceRequestsOOv1];
-
-  console.log(`Checking proposal price for ${livePolymarketProposalRequests.length} markets...`);
-
-  const notifiedProposals = [];
-  for (const market of livePolymarketProposalRequests) {
-    if (Object.keys(pastNotifiedProposals).includes(getMarketKeyToStore(market))) continue;
-    try {
-      const processingResult = await processMarketProposal(market, params, logger);
-      if (processingResult.notified) notifiedProposals.push(market);
-    } catch (error) {
-      await logFailedMarketProposalVerification(logger, params.chainId, market, error as Error);
-      notifiedProposals.push(market);
-    }
-  }
-
-  await storeNotifiedProposals(notifiedProposals);
-  console.log("All proposals have been checked!");
+// Retrieve threshold values from environment variables.
+function getThresholds() {
+  return {
+    asks: Number(process.env["THRESHOLD_ASKS"]) || 1,
+    bids: Number(process.env["THRESHOLD_BIDS"]) || 0,
+    volume: Number(process.env["THRESHOLD_VOLUME"]) || 500000,
+  };
 }
 
-async function processMarketProposal(market: OptimisticPriceRequest, params: MonitoringParams, logger: typeof Logger) {
+async function processProposal(proposal: OptimisticPriceRequest, params: MonitoringParams, logger: typeof Logger) {
   const networker = new Networker(logger);
-  const questionID = calculatePolymarketQuestionID(market.ancillaryData);
-  // set this to retry twice and wait 5 seconds between retries.
-  const polymarketInfo = await retryAsync(
+  const isSportsMarket = proposal.requester === params.ctfSportsOracleAddress;
+  const questionID = calculatePolymarketQuestionID(proposal.ancillaryData);
+
+  // Retry fetching market information per configuration.
+  const markets = await retryAsync(
     () => getPolymarketMarketInformation(logger, params, questionID),
     params.retryAttempts,
     params.retryDelayMs
   );
-  const orderBook = await getPolymarketOrderBook(params, polymarketInfo.clobTokenIds, networker);
-  const orderFilledEvents = await getOrderFilledEvents(
-    params,
-    polymarketInfo.clobTokenIds,
-    Number(market.requestBlockNumber)
-  );
 
-  const proposedOutcome = market.proposedPrice.eq(ONE_SCALED) ? 0 : 1;
-  const complementaryOutcome = proposedOutcome === 0 ? 1 : 0;
+  for (const marketInfo of markets) {
+    let scores: [ethers.BigNumber, ethers.BigNumber] = [ethers.BigNumber.from(0), ethers.BigNumber.from(0)];
+    let multipleValuesQuery: MultipleValuesQuery | undefined;
+    let winnerOutcome: number;
+    let loserOutcome: number;
 
-  const thresholds = {
-    asks: Number(process.env["THRESHOLD_ASKS"]) || 1,
-    bids: Number(process.env["THRESHOLD_BIDS"]) || 0,
-    volume: Number(process.env["THRESHOLD_VOLUME"]) || 500000,
-  };
+    if (isSportsMarket) {
+      // Skip if the proposed price is unresolvable.
+      if (isUnresolvable(proposal.proposedPrice)) {
+        continue;
```

Comment:
Skip processing if the proposed price is unresolvable as the order book cannot be reliably interpreted